src/Model/ResourceField.php

```diff
@@ -58,6 +60,15 @@ public function getCMSFields()
             $fields->removeByName('Type');
             $fields->dataFieldByName('ReadableName')
                 ->setAttribute('placeholder', $this->Name);
+            $orderField = NumericField::create('Order')
+                ->setTitle(i18n::_t(__CLASS__ . '.OrderLabel', 'Presented order'))
+                ->setDescription(i18n::_t(
+                    __CLASS__ . '.OrderDenominator',
+                    'of {count} fields',
+                    ['count' => static::get()->filter('ResourceID', $this->ResourceID)->count()]
```

Comment:
Wouldn't `$this->Resource->Fields()->count()` potentially be more performant?

Reply:
Resource is a query in itself I believe, since it's has_one. Which makes 2 queries.
You'd be right if it returned a DataList as *many relationships do.

I'll double check my thinking :)

Reply:
Ok sounds fair. I was hoping that SilverStripe ORM would be smart enough to populate the parent resource on a child field when it loads it but that doesn't seem to be the case. New hackday ideas 🤔